Semantic Scholar.js

```diff
@@ -122,21 +129,37 @@ function parseDocument(doc, url) {
 			item.issue = volumeAndIssue[1];
 		}
 		
-		if (rawData.hasPdf && (rawData.primaryPaperLink.linkType === 's2'
-			|| rawData.primaryPaperLink.linkType == 'arxiv')) {
-			item.attachments.push({
-				url: rawData.primaryPaperLink.url,
-				title: "Full Text PDF",
-				mimeType: 'application/pdf'
-			});
-			if (rawData.primaryPaperLink.linkType == 'arxiv') {
-				let arxivId = rawData.primaryPaperLink.url.match(/\d{4}\.\d{5}/);
-				if (arxivId.length >= 1) {
-					if (item.extra) {
-						item.extra += '\narXiv: ' + arxivId[0];
+		if (rawData.primaryPaperLink) {
+			let pdfLinkElement;
+			if (rawData.primaryPaperLink.url.endsWith('.pdf')) {
+				pdfLinkElement = rawData.primaryPaperLink;
+			}
+			else if (rawData.alternatePaperLinks) {
+				for (let i = 0; i < rawData.alternatePaperLinks.length; i++) {
```

Comment:
```suggestion
				for (let alternateElement of rawData.alternatePaperLinks) {
```

Reply:
Then the following line is not anymore needed (and you never use the variable `i` anyways here), so this is a simplification of your code.